homeassistant/components/input_boolean/__init__.py

```diff
@@ -92,7 +92,7 @@ def is_on(hass: HomeAssistant, entity_id: str) -> bool:
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up an input boolean."""
-    component = EntityComponent(_LOGGER, DOMAIN, hass)
+    component = EntityComponent[InputBoolean](_LOGGER, DOMAIN, hass)
```

Comment:
In most of these cases the specialization doesn't change much. Since the generic for `EntiyComponent` has such a limited use case, my suggestion would have been to only add it where it can help with typing. I.e. `automation/__init__.py`. The remaining cases could have been left empty. At some point [PEP 696](https://peps.python.org/pep-0696/) might get accepted which would help further.

**Edit**: Just saw the mypy error. That shouldn't raise `var-annotated`, only `type-arg` if `disallow_any_generics` is set. Anyway 🤷🏻‍♂️

Reply:
Try searching for `component.get_entity` to see where it would be useful.